### DISCUSSION TITLE
Is it possible to support newline characters in an error message?

### DISCUSSION THREAD

**Discussion by LevinGermann** [directus/directus#17501] [Archive]
I want to do some server-side validation before a user can save an item from a given collection.
For that I'm using a filter flow using a web request (with the operation-key `request`), with a "Run Script" operation at the end that looks like this:

```
module.exports = async function(data) {
    let affectedSlots = data.request.data.slots;
    if(affectedSlots.length > 0){
        let slotsString = affectedSlots.reduce((acc, slot) => {
            return acc + slot.bookingId + ' -> ' + slot.date + ' ' + slot.time + '\n / ';
        }, '');
        throw new Error(`The following slot(s) would be affected by this closure: \n${slotsString}`); 
    }else{
       return {}; 
    }
}
```

This snippet leads to this error message:
![image](https://user-images.githubusercontent.com/9347206/218505880-ec3e999a-235c-4c43-8073-b175e307db9b.png)

Is it possible to display this WITH the newline characters? If I copy the error message, the `\n` characters are there.
Or is there another recommended approach on how I can tell a user the reason why they can't save?

Cheers! 😊

**Comment by rijkvanzanten** [comment]
Do `\n` characters work as expected within a backtick string? 🤔 

I don't think the v-error component that renders that has any special CSS preventing the linebreaks

**Reply by LevinGermann** [reply]
I tried all these approaches, unfortunately without success. 😞 

Regular tick:
`throw new Error('Example \n Newline');`

<img width="519" alt="image" src="https://user-images.githubusercontent.com/9347206/218986715-9f3f7835-e98b-4f1c-9337-85b4e0baeedd.png">

```
{
  "message": "Example \n Newline",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}
```

----

Backtick:
```throw new Error(`Example \n Newline`);```

<img width="515" alt="image" src="https://user-images.githubusercontent.com/9347206/218987498-8e359183-1dd1-4bf5-8cc2-da86b7d2a9d9.png">

```
{
  "message": "Example \n Newline",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}
```

----

Escaping:
```throw new Error('Example \\n Newline');```

<img width="515" alt="image" src="https://user-images.githubusercontent.com/9347206/218987798-d49e84b4-3e42-4e9b-8f42-edfe5285e2e7.png">

```
{
  "message": "Example \\n Newline",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}
```

**Reply by br41nslug** [reply]
I don't think the `<output>` tag used renders newlines (that are not `<br>`) for that we'd have to add a style like `white-space: pre;`

**Reply by LevinGermann** [reply]
@br41nslug You're right, I had success with adding the `white-space: pre-line` property to the output element:

```throw new Error(`1 \n 2`);```
![image](https://user-images.githubusercontent.com/9347206/218994170-b38cd824-a805-496e-b838-ad6f4581632d.png)
```
{
  "message": "1 \n 2",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}
```

Not sure what other implications this has, but for this use-case this seems to be working

**Reply by LevinGermann** [reply]
For what it's worth, just using `<br>` tags (without any CSS adjustments) isn't rendered like a line break:

![image](https://user-images.githubusercontent.com/9347206/218994797-462f2bb6-7fe0-416d-a7ea-5124fae37cdb.png)

**Reply by br41nslug** [reply]
Thats more likely due to Vue/Directus escaping the HTML in the string than that the browser wouldnt render it. If you we're to add the `<br>` directly in the HTML it should break the line.

> I don't think the v-error component that renders that has any special CSS preventing the linebreaks

A lot of html elements require extra CSS to respect regular line breaks ;)

**Reply by LevinGermann** [reply]
<img width="927" alt="image" src="https://user-images.githubusercontent.com/9347206/218998259-047aaf18-e562-4564-a598-fe52536f21d8.png">

🤔

**Reply by br41nslug** [reply]
Looks like it was pasted as text not as html (because the dev panel color highlights html code). in my tests it works in chrome and firefox ;) But that hardly matters for the original issue

**Reply by LevinGermann** [reply]
Whoops - you're right! Sorry for the confusion 😄

**Reply by rijkvanzanten** [reply]
> I don't think the `<output>` tag used renders newlines (that are not `<br>`) for that we'd have to add a style like `white-space: pre;`

Ah good one. Didn't realize it was in an `<output>` 👍🏻
